packages/delisp-core/src/unify.ts

```diff
@@ -226,7 +239,16 @@ export function unify(
         };
   } else if (t1.type === "application" && t2.type === "application") {
     // RULE: (uni-app)
-    return unifyArray(t1.args, t2.args, ctx);
+    const argResult = unifyArray(
```

Comment:
Is this the result of some previous attempts to fix it?  I don't see how this differs from just a call to unifyArray.

Reply:
Oh sorry, I just saw the PR description 👍 sounds like a good idea!